### ISSUE TITLE
Planning migration to v0.14 API

### ISSUE THREAD

**Issue by y-ken** [y-ken/fluent-plugin-pagerduty#4]
Hello @okkez
I'm afraid could you please revise up-to-date with latest fluentd?
It should be better to follow latest fluentd api.

There is a user in this plugin
https://github.com/y-ken/fluent-plugin-pagerduty/pull/3

**Comment by okkez** [comment]
Sure, I'm trying.
I have some questions:
* [event_type](https://github.com/y-ken/fluent-plugin-pagerduty/blob/60219f92864104b7036c5d4bd3ceb01f9626bf6e/lib/fluent/plugin/out_pagerduty.rb#L49) is assigned but not used
  * Do you have any plan to use `event_type`?
* [incident](https://github.com/y-ken/fluent-plugin-pagerduty/blob/60219f92864104b7036c5d4bd3ceb01f9626bf6e/lib/fluent/plugin/out_pagerduty.rb#L64) is assigned but not used
  * Do you have any plan to use `incident`?
* Do you want to replace PlaceholderExpander with built-in `extract_placeholders`?
  * I want to use built-in methods to reduce code

I found that we cannot start incident_key/description with `[` (left square bracket) and `{` (left wave bracket). So we need updating example in README.

**Comment by okkez** [comment]
Hmm, we cannot use built-in `extract_placeholders` in non-buffered output plugin, for now.

**Comment by y-ken** [comment]
Hi @okkez
Thank you for your quick action!

I'll confirm event_type and incident tomorrow.
And, built-in extract_placeholders is better. And also It needs to be buffered plugin.
In the case of changing buffered plugin, Please set `flush_interval: 0` like following
https://github.com/y-ken/fluent-plugin-geoip/blob/master/lib/fluent/plugin/out_geoip.rb

**Comment by okkez** [comment]
@y-ken Any progress?

**Comment by y-ken** [comment]
Hi @okkez

I'm sorry for my read pager duty api again delayed.
These are unused and useless variables.

* `event_type` is useless. it can erase the line 46. And It kicks only trigger mode with description. So, It need check only @description.nil? condition at Line 23
* `incident` does not need to assign return value.

**Comment by okkez** [comment]
Thank you for reading code and I've updated #5.

I have additional questions.

Does this plugin support `event_type` parameter "acknowledge" and "resolve"?
It seems that this plugin does not support those event types.
Because we don't call APIs for those event types in this plugin.

Do you want to support all event types by this plugin?
I think that we don't need to support all event types in this plugin for now. Because we cannot acknowledge and resolve the triggered events automatically without defining rules to acknowledge or resolve.

**Comment by y-ken** [comment]
Yes exactly. We don't need to support all event types for now. #5 seems fine. Great!
Do you think is it better to introduce error handling in the case of coming "acknowledge" and "resolve"?

**Comment by okkez** [comment]
How about following? Drop both "acknowledge" and "resolve" for now .

```ruby
  # TODO support :acknowledge, :resolve
  config_param :event_type, :enum, list: [:trigger], default: :trigger
```
